File: sdks/go/pkg/models/operations/getshareditemsnumberweblinks.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetSharedItemsNumberWebLinksQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type GetSharedItemsNumberWebLinksHeaders struct {
	Boxapi      string  `header:"style=simple,explode=false,name=boxapi"`
	IfNoneMatch *string `header:"style=simple,explode=false,name=if-none-match"`
}

type GetSharedItemsNumberWebLinksRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetSharedItemsNumberWebLinksQueryParams
	Headers     GetSharedItemsNumberWebLinksHeaders
}

type GetSharedItemsNumberWebLinksResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	WebLink     *shared.WebLink
}

File: sdks/ts/src/sdk/models/operations/getshareditemsnumberweblinks.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetSharedItemsNumberWebLinksQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class GetSharedItemsNumberWebLinksHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=boxapi" })
  boxapi: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=if-none-match" })
  ifNoneMatch?: string;
}


export class GetSharedItemsNumberWebLinksRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetSharedItemsNumberWebLinksQueryParams;

  @SpeakeasyMetadata()
  headers: GetSharedItemsNumberWebLinksHeaders;
}


export class GetSharedItemsNumberWebLinksResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  webLink?: shared.WebLink;
}
